src/main/groovy/nextflow/cwl/CwlRunner.groovy

```diff
@@ -46,20 +46,38 @@ class CwlRunner {
 
         def cmd = new StringBuilder()
         // -- base command
-        if( yaml.baseCommand )
-            cmd << yaml.baseCommand
+        if( yaml.baseCommand ) {
+            if ( yaml.baseCommand.class == String )
+                cmd << yaml.baseCommand
+            else if ( yaml.baseCommand.class == ArrayList )
+                cmd << yaml.baseCommand.join(' ')
+            else
+                throw new IllegalArgumentException("CWL unknown baseCommand format")
+
+        }
         else
             throw new IllegalArgumentException("Not a valid CWL descriptor -- Missing `baseCommand`")
-
         // -- command arguments
         def args = (List)yaml.arguments
         if( args )
             cmd << ' ' << args.join(' ')
 
+        if( yaml.hints ){
+            if( yaml.hints.DockerRequirement )
+                config.container = yaml.hints.DockerRequirement.dockerPull
+        }
+
```

Comment:
This can be simplified as 

```
if( yaml.hints?.DockerRequirement ) {
    config.container = yaml.hints.DockerRequirement.dockerPull
}
```

Note: the question mark (safe property access operator)